multicluster/service-mirror/cluster_watcher.go

```diff
@@ -192,18 +192,30 @@ func (rcsw *RemoteClusterServiceWatcher) originalResourceName(mirroredName strin
 	return strings.TrimSuffix(mirroredName, fmt.Sprintf("-%s", rcsw.link.TargetClusterName))
 }
 
-func (rcsw *RemoteClusterServiceWatcher) getMirroredServiceLabels() map[string]string {
-	return map[string]string{
-		consts.MirroredResourceLabel:  "true",
-		consts.RemoteClusterNameLabel: rcsw.link.TargetClusterName,
+func (rcsw *RemoteClusterServiceWatcher) getMirroredServiceLabels(remoteService *corev1.Service) map[string]string {
+	labels := make(map[string]string)
+
+	if remoteService != nil {
```

Comment:
nit: we could probably get rid of some of the nesting if we check for nil equivalence here instead. We'd have to move up the multicluster labels up a bit though. You can also initialise the map directly with these labels if you want.
e.g
```go
labels := map[string]string{
   costs.MirroredResourceLabel: "true",
   costs.RemoteClusterNameLabel: rcsw.link.TargetClusterName
}

if remoteService == nil {
   return labels
}

for (...)

return labels
}
```

Reply:
Cool. Other than readability, does this also affect memory? If it's better, we can do the same for annotations as well.

Reply:
Hmm, it doesn't affect memory, no. I spent a bit of time looking into this, but [composite literals](https://golang.org/ref/spec#Composite_literals) shouldn't have an effect on memory, at least not in the case of maps. This is the same as doing everything separately, I think it looks a bit cleaner this way (it also provides a reasonable default, especially in a function where we take an "optional" parameter).

As far as memory is concerned, maps are typically very optimised to grow beyond their capacity bounds. If we'd want to micro-optimise this, we could probably do something like:
```go
labels := make(map[string]string, len(remoteService.Labels) + 2)) // where +2 comes from our default annotations
labels[consts.MirroredResourceLabel] = "true"
(...)
```

We'd essentially set a "capacity" hint so that we force as fewer operations on the underlying map (it wouldn't have to grow itself, not sure what the default is tbh). This would have the drawback of making everything a bit more complicated, we also wouldn't really gain a lot from it.

So my TL;DR is -- it doesn't affect memory, it makes it easier to read and provides a sensible default so we can make better assumptions about what's happening to the code (at least from my pov).